submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireService.kt

```diff
@@ -40,40 +34,38 @@ class FireService(
      * different path) and if so, even if the file exists in FIRE, it gets duplicated to ensure consistency.
      */
     fun getOrPersist(sub: ExtSubmission, file: ExtFile): FirePersistResult = when (file) {
-        is FireFile -> fromFireFile(sub, file, "${sub.relPath}/${file.relPath}")
+        is FireFile -> fromFireFile(file, "${sub.relPath}/${file.relPath}")
         is NfsFile -> when (file.type) {
-            FILE -> fromNfsFile(sub, file, "${sub.relPath}/${file.relPath}")
-            DIR -> fromNfsFile(sub, file.copy(file = compress(sub, file.file)), "${sub.relPath}/${file.relPath}.zip")
+            FILE -> fromNfsFile(file, "${sub.relPath}/${file.relPath}")
+            DIR -> fromNfsFile(file.copy(file = compress(sub, file.file)), "${sub.relPath}/${file.relPath}.zip")
         }
     }
 
-    private fun fromNfsFile(sub: ExtSubmission, file: NfsFile, expectedPath: String): FirePersistResult =
-        reuseOrPersistFireFile(sub, file, expectedPath) { file.file }
+    private fun fromNfsFile(file: NfsFile, expectedPath: String): FirePersistResult =
+        reuseOrPersistFireFile(file, expectedPath) { file.file }
 
-    private fun fromFireFile(sub: ExtSubmission, file: FireFile, expectedPath: String): FirePersistResult =
-        reuseOrPersistFireFile(sub, file, expectedPath) { client.downloadByFireId(file.fireId, file.fileName) }
+    private fun fromFireFile(file: FireFile, expectedPath: String): FirePersistResult =
+        reuseOrPersistFireFile(file, expectedPath) { client.downloadByFireId(file.fireId, file.fileName) }
 
     @Suppress("ReturnCount")
     private fun reuseOrPersistFireFile(
-        sub: ExtSubmission,
         file: ExtFile,
         expectedPath: String,
         fallbackFile: () -> File,
     ): FirePersistResult {
-        val fireFile = client.findByMd5(file.md5).firstOrNull { it.isAvailable(sub.accNo) }
-        if (fireFile != null) {
-            when (fireFile.path) {
-                null -> return FirePersistResult(setMetadata(sub, fireFile.fireOid, file, expectedPath), false)
-                expectedPath -> return FirePersistResult(asFireFile(file, fireFile.fireOid), false)
-            }
-        }
+        val files = client.findByMd5(file.md5)
+
+        val byPath = files.firstOrNull { it.filesystemEntry?.path == expectedPath }
+        if (byPath != null) return FirePersistResult(asFireFile(file, byPath.fireOid), false)
+
+        val noPath = files.firstOrNull { it.filesystemEntry?.path == null }
```

Comment:
I think this one should go first since it's the most likely to happen because it's supposed that we cleaned everything in the previous step